sklearn/datasets/twenty_newsgroups.py

```diff
@@ -345,8 +346,8 @@ def fetch_20newsgroups_vectorized(subset="train", remove=(), data_home=None):
 
     # the data is stored as int16 for compactness
     # but normalize needs floats
-    X_train = X_train.astype(np.float64)
-    X_test = X_test.astype(np.float64)
+    X_train = astype(X_train, np.float64, copy=False)
+    X_test = astype(X_test, np.float64, copy=False)
```

Comment:
There is no need to set `copy=False` for this use case as the original data type is `int16` for sure as explained in the previous commit. There is no way to avoid the malloc / dtype conversion in that case. We can keep:

```python
    X_train = X_train.astype(np.float64)
    X_test = X_test.astype(np.float64)
```